src/snowflake/connector/auth/by_plugin.py

```diff
@@ -210,12 +210,13 @@ def handle_timeout(
         del authenticator, service_name, account, user, password
         logger.debug("Default timeout handler invoked for authenticator")
         if not self._retry_ctx.should_retry():
-            self._retry_ctx.reset()
-            raise OperationalError(
+            error = OperationalError(
                 msg=f"Could not connect to Snowflake backend after {self._retry_ctx.get_current_retry_count()} attempt(s)."
                 "Aborting",
                 errno=ER_FAILED_TO_CONNECT_TO_DB,
             )
+            self._retry_ctx.reset()
+            raise error
```

Comment:
The change in this file seems like a no-op? It probably makes it easier to step through the code during debugging though so it's okay to keep.

Reply:
In the original issue, it was pointed out that when `_retry_ctx.reset()` is called before the error message is created, the message does not reflect the correct number of retries - it gets reset to 0. Which is why we are changing this order

Reply:
Ah I see. Thanks for the clarification